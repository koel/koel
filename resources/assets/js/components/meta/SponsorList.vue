<template>
  <div class="sponsors">
    <a
      v-for="sponsor in sponsors"
      :key="sponsor.url"
      :href="sponsor.url"
      :title="sponsor.description"
      target="_blank"
    >
      <img :alt="sponsor.description" :src="sponsor.logo.src" :style="sponsor.logo.style">
    </a>
  </div>
</template>

<script lang="ts" setup>
import sponsors from '@/sponsors'</script>

<style lang="scss" scoped>
.sponsors {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  justify-content: center;
  row-gap: 4px;
  column-gap: 16px;
  padding: 8px;
  background: rgba(0, 0, 0, .1);
  border-radius: 5px;
  border: 1px solid rgba(255, 255, 255, .05);

  img {
    filter: brightness(10);
    height: 32px;
  }
}
</style>
