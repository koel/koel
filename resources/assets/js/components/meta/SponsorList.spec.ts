import { expect, it } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import SponsorList from './SponsorList.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders', () => {
      expect(this.render(SponsorList).html()).toMatchSnapshot()
    })
  }
}
