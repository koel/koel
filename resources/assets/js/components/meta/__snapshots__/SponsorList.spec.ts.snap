// Vitest Snapshot v1

exports[`renders 1`] = `<div class="sponsors" data-v-46e7cf66=""><a href="https://render.com" title="Render - Cloud Hosting for Developers" target="_blank" data-v-46e7cf66=""><img alt="Render - Cloud Hosting for Developers" src="undefined/resources/assets/img/sponsors/render.svg" style="height: 28px;" data-v-46e7cf66=""></a><a href="https://www.keycdn.com?a=11519" title="KeyCDN - Content delivery made easy" target="_blank" data-v-46e7cf66=""><img alt="KeyCDN - Content delivery made easy" src="undefined/resources/assets/img/sponsors/keycdn.svg" data-v-46e7cf66=""></a><a href="https://whatthediff.ai" title="What The Diff - AI powered changelog generation" target="_blank" data-v-46e7cf66=""><img alt="What The Diff - AI powered changelog generation" src="undefined/resources/assets/img/sponsors/what-the-diff.svg" style="height: 20px;" data-v-46e7cf66=""></a></div>`;
